stable_baselines3/common/vec_env/vec_normalize.py

```diff
@@ -50,9 +51,27 @@ def __init__(
             if isinstance(self.observation_space, gym.spaces.Dict):
                 self.obs_spaces = self.observation_space.spaces
                 self.obs_rms = {key: RunningMeanStd(shape=self.obs_spaces[key].shape) for key in self.norm_obs_keys}
+                # Update observation space when using image
+                for key in self.obs_rms.keys():
+                    if is_image_space(self.obs_spaces[key]):
+                        self.observation_space.spaces[key] = gym.spaces.Box(
+                            low=-clip_obs,
+                            high=clip_obs,
+                            shape=self.obs_spaces[key].shape,
+                            dtype=np.float32,
+                        )
+
             else:
                 self.obs_spaces = None
                 self.obs_rms = RunningMeanStd(shape=self.observation_space.shape)
+                # Update observation space when using image
+                if is_image_space(self.observation_space):
+                    self.observation_space = gym.spaces.Box(
+                        low=-clip_obs,
+                        high=clip_obs,
+                        shape=self.observation_space.shape,
+                        dtype=np.float32,
+                    )
```

Comment:
Why that applies only for images?

Reply:
we could apply it for others, but this would create issues with already saved models (we check the obs space at load time).
In the case here, it's to raise proper error when using image-like input with vecnormalize.
For other cases, Vecnormalize is meant to be used with Box space only, so we would only change the bounds (i agree it would be more correct but I'm afraid it would create more issues).
I'm happy to change my mind though ;)

Reply:
I understand. I'm in two minds about it. Maybe the best thing for now is to stay like this and add a comment stating what you just explained to me, to potentially help a later improvement.